packages/clay-radio-group/src/ClayRadioGroup.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface RadioProps extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Props for the outer most container element.
+	 */
+	containerProps?: React.HTMLAttributes<HTMLDivElement>;
+
+	/**
+	 * Flag to indicate if radio elements should display inline.
+	 */
+	inline?: boolean;
+
+	/**
+	 * Text to describe for radio element.
+	 */
+	label?: React.ReactText;
+
+	/**
+	 * Value provided if element is selected.
+	 */
+	value: React.ReactText;
+}
+
+type RadioType = React.FunctionComponent<RadioProps>;
+
+const Radio: RadioType = ({
+	checked,
+	children,
+	className,
+	containerProps = {className: ''},
+	inline,
+	label,
+	...otherProps
+}) => {
+	return (
+		<div
+			{...containerProps}
+			className={classNames(
+				'custom-control custom-radio',
+				containerProps.className,
+				{
+					'custom-control-inline': inline,
+				}
+			)}
+		>
+			<label>
+				<input
+					{...otherProps}
+					checked={checked}
+					className={classNames('custom-control-input', className)}
+					role="radio"
+					type="radio"
+				/>
+
+				<span className="custom-control-label">
+					{label && (
+						<span className="custom-control-label-text">
+							{label}
+						</span>
+					)}
+				</span>
+
+				{children}
+			</label>
+		</div>
+	);
+};
+
+interface GroupProps extends React.HTMLAttributes<HTMLDivElement> {
+	children: React.ReactElement<RadioProps>[];
+
+	/**
+	 * Flag to indicate if radio elements should display inline.
+	 */
+	inline?: boolean;
+
+	/**
+	 * Form element `name` that is applied to each radio element.
+	 */
+	name?: string;
+
+	/**
+	 * Callback function for whenever a radio element is selected.
+	 */
+	onSelectedValueChange: (val: React.ReactText) => void;
+
+	/**
+	 * The value that corresponds to the selected radio element. Leave
+	 * undefined if no option is selected.
+	 */
+	selectedValue?: React.ReactText;
+}
+
+const ClayRadioGroup: React.FunctionComponent<GroupProps> & {
+	Radio: RadioType;
+} = ({
+	children,
+	className,
+	inline,
+	name,
+	onSelectedValueChange,
+	selectedValue,
+	...otherProps
+}) => {
+	return (
+		<div {...otherProps} className={classNames(className)}>
+			{children.map((child, i) => {
```

Comment:
It may be safer to use the `React.Children.map` API.

Reply:
I was looking at the react docs and it seemed like `React.Children.map` was to be used specifically when you want to pass the `this` instance to each child. Maybe I'm wrong though, did you have a different understanding of its purpose?

Reply:
I think the documentation passes a different feeling about its use, but we can pass a function and clone the element. I say it may be safer because it traverse the children doing some treatments, I think looking at the code looks better. https://github.com/facebook/react/blob/master/packages/react/src/ReactChildren.js#L311-L346

Reply:
I think the idea is that "children" is actually an opaque data structure so the `React.Children` utils isolate you from having to worry about whether you have an array, an array of arrays, a string etc. Funnily enough, I think the [Flow documentation on Children](https://flow.org/en/docs/react/children/) might actually be the best summary of the kinds of shapes that children can take.

Reply:
Oh thanks @wincent for the info.